src/Tag/Storage/DoctrineCache.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace FOS\HttpCache\Tag\Storage;
+
+use Doctrine\Common\Cache\Cache;
+use FOS\HttpCache\Tag\StorageInterface;
+
+/**
+ * Tag storage implementation which uses the Doctrine Cache library.
+ */
+class DoctrineCache implements StorageInterface
+{
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    public function __construct(Cache $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    /**
+     * TODO: The lifetime applies not to the tag but to the identifier - we
+     *       need to serialize the lifetime with the cache entry and store tags
+     *       indefintely.
+     *
+     * {@inheritDoc}
+     */
+    public function tagCacheId(array $tags, $identifier, $lifetime = null)
```

Comment:
Do we need to store the lifetime of the content/cache response?

We can just have a command which iterates over the cache entries and removes the entries which no longer have a valid cache ID.